src/pybamm/config.py

```diff
@@ -45,20 +48,125 @@ def is_running_tests():  # pragma: no cover
+
+    Returns:
+        str | None: The user input if received before the timeout, or None if:
+            - The timeout was reached
+            - Telemetry is disabled via environment variable
+            - Running in a non-interactive environment
+            - An error occurred
+
+        The caller can distinguish between an empty response (user just pressed Enter)
+        which returns '', and a timeout/error which returns None.
+    """
+    # Check for telemetry disable flag
+    if os.getenv("PYBAMM_DISABLE_TELEMETRY", "false").lower() != "false":
+        return None
+
+    if not (sys.stdin.isatty() or is_notebook()):
+        return None
 
-    Parameters
-    ----------
-    timeout : float, optional
-        The timeout for the user to respond to the prompt. Default is 10 seconds.
+    # 1. Handling for Jupyter notebooks. This is a simplified
+    # implementation in comparison to widgets because notebooks
+    # are usually run interactively, and we don't need to worry
+    # about the event loop. The input is disabled when running
+    # tests due to the environment variable set.
+    if is_notebook():  # pragma: no cover
+        try:
+            from IPython.display import clear_output
+
+            user_input = input("Do you want to enable telemetry? (Y/n): ")
+            clear_output()
+
+            return user_input
+
+        except Exception:  # pragma: no cover
+            return None
+
+    # 2. Windows-specific handling
+    if sys.platform == "win32":
+        try:
+            import msvcrt
+
+            start_time = time.time()
+            input_chars = []
+            sys.stdout.write("Do you want to enable telemetry? (Y/n): ")
+            sys.stdout.flush()
+
+            while time.time() - start_time < timeout:
+                if msvcrt.kbhit():
+                    char = msvcrt.getwche()
+                    if char in ("\r", "\n"):
+                        sys.stdout.write("\n")
+                        return "".join(input_chars)
+                    input_chars.append(char)
+                time.sleep(0.1)
+            return None
+        except Exception:
+            return None
 
-    Returns
-    -------
-    bool
-        True if the user opts in, False otherwise.
+    # 3. POSIX-like systems will need to use termios
+    else:  # pragma: no cover
+        try:
+            import termios
+            import tty
+            import select
+
+            # Save terminal settings for later
+            old_settings = termios.tcgetattr(sys.stdin)
+            try:
+                # Set terminal to raw mode
+                tty.setraw(sys.stdin.fileno())
+
+                sys.stdout.write("Do you want to enable telemetry? (Y/n): ")
+                sys.stdout.flush()
+
+                input_chars = []
+                start_time = time.time()
+
+                while time.time() - start_time < timeout:
+                    rlist, _, _ = select.select([sys.stdin], [], [], 0.1)
```

Comment:
If we are already using select, do we need the loop? The timeout isn't very long